pandas/core/indexes/datetimes.py

```diff
@@ -532,9 +528,13 @@ def union_many(self, others):
             else:
                 tz = this.tz
                 this = Index.union(this, other)
-                if isinstance(this, DatetimeIndex):
-                    this._tz = timezones.tz_standardize(tz)
-
+                if isinstance(this, DatetimeIndex) and tz is not None:
```

Comment:
this is what the ._tz setter used to do, so just make a method to do instead (as you are calling in multiple places), maybe ``_set_timezone``

Reply:
OK.  I'm kind of hoping this is all unnecessary, will take a look

Reply:
ok its fine for now.

Reply:
just pushed with a fix here: just set `self._eadata._dtype`, analogous to the previous method setting `self._tz`.

Although in all of the tests where this is hit, it already has the appropriate dtype, so this may be superfluous.

Reply:
k